src/MeshTransfer.f90

```diff
@@ -14,34 +14,848 @@
+        CASE('CONTINUOUS')
+          this%MapType_in=MapType_CONTINUOUS
+        CASE('POINT')
+          this%MapType_in=MapType_POINT
+        CASE('VOLUME')
+          this%MapType_in=MapType_VOLUME
+        CASE DEFAULT
+          WRITE(*,*) "invalid incomming map type"
+          STOP -3
+      ENDSELECT
+      IF(this%MapType_in == MapType_CONTINUOUS) THEN
+        CALL pList%get('moments_in',this%nmesh_in)
+      ELSEIF(this%MapType_in == MapType_POINT) THEN
+        CALL pList%get('pointmesh_in',this%mesh_in)
+        this%nmesh_in=SIZE(this%mesh_in)
+      ELSEIF(this%MapType_in == MapType_VOLUME) THEN
+        CALL pList%get('volumemesh_in',this%mesh_in)
+        this%nmesh_in=SIZE(this%mesh_in)-1
+      ENDIF
+
+      CALL pList%get('map_out',tmpstr)
+      SELECTCASE(CHAR(tmpstr))
+        CASE('CONTINUOUS')
+          this%MapType_out=MapType_CONTINUOUS
+        CASE('POINT')
+          this%MapType_out=MapType_POINT
+        CASE('VOLUME')
+          this%MapType_out=MapType_VOLUME
+        CASE DEFAULT
+          WRITE(*,*) "invalid outgoing map type"
+          STOP -3
+      ENDSELECT
+      IF(this%MapType_out == MapType_CONTINUOUS) THEN
+        CALL pList%get('moments_out',this%nmesh_out)
+      ELSEIF(this%MapType_out == MapType_POINT) THEN
+        CALL pList%get('pointmesh_out',this%mesh_out)
+        this%nmesh_out=SIZE(this%mesh_out)
+      ELSEIF(this%MapType_out == MapType_VOLUME) THEN
+        CALL pList%get('volumemesh_out',this%mesh_out)
+        this%nmesh_out=SIZE(this%mesh_out)-1
+      ENDIF
+
+      IF(pList%has('poly_transfer')) THEN
+        CALL pList%get('poly_transfer',this%poly_transfer)
+      ENDIF
+
+      IF(this%MapType_in /= MapType_CONTINUOUS) THEN
+        DO i=2,SIZE(this%mesh_in)
+          ENSURE(this%mesh_in(i)>this%mesh_in(i-1))
+        ENDDO
+      ENDIF
+      IF(this%MapType_out /= MapType_CONTINUOUS) THEN
+        DO i=2,SIZE(this%mesh_out)
+          ENSURE(this%mesh_out(i)>this%mesh_out(i-1))
+        ENDDO
+      ENDIF
+
+    ENDSUBROUTINE init_1Dbase
+!
+!-------------------------------------------------------------------------------
+!> @brief
+!> @param this  The mesh transfer object
+!>
+!> This routine
+!>
+    SUBROUTINE clear_1Dbase(this)
+      CLASS(MeshTransfer_1Dbase),INTENT(INOUT) :: this
+
+      this%MapType_in=0
+      this%MapType_out=0
+      this%nmesh_in=0
+      this%nmesh_out=0
+      this%poly_transfer=-1
+      IF(ALLOCATED(this%mesh_in)) DEALLOCATE(this%mesh_in)
+      IF(ALLOCATED(this%mesh_out)) DEALLOCATE(this%mesh_out)
+      IF(ALLOCATED(this%transferMatrix)) DEALLOCATE(this%transferMatrix)
+      IF(this%transferLS%isInit) CALL this%transferLS%clear()
+
+      this%isInit=.FALSE.
+    ENDSUBROUTINE clear_1Dbase
+!
+!-------------------------------------------------------------------------------
+!> @brief
+!> @param this  The mesh transfer object
+!>
+!> This routine
+!>
+    SUBROUTINE transfer_1Dbase(this,data_in,data_out)
```

Comment:
For type bound procedures, I'm generally not a fan of the passed object being called ```this```. It is not descriptive enough. In particular when you have a single file defining multiple concrete types, if the routine is long enough that you can be looking in the middle of it and not see the encapsulating construct or definition of ```this```, then it's not always obvious what type ```this``` is.

I think it's much better to have ```thisXXX``` where ```XXX``` is some acronym for the type. In this case (1DB, 1DC or 1DR).